### PR TITLE
Kafka write metrics and tests

### PR DESCRIPTION
**Pull request by nickuncaged1201**
Add error metrics and dlq to Kafka write schema transform.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by nickuncaged1201** [comment]
r: @johnjcasey @Dippatel98 @manavgarg @pabloem

**Review by manavgarg — APPROVED** [review]
I think the PR name is incorrect. Shouldn't it be "Kafka write metrics and tests" ?
LGTM otherwise.

File: sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaWriteSchemaTransformProvider.java [hunk 1]
```diff
@@ -99,13 +144,21 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
...
-              .get("input")
-              .apply(
-                  "Map Rows to Kafka Messages",
-                  MapElements.via(
-                      new SimpleFunction<Row, KV<byte[], byte[]>>(
-                          row -> KV.of(new byte[1], toBytesFn.apply(row))) {}))
+          PCollectionTuple outputTuple =
+              input
+                  .get("input")
+                  // .apply(
```
Comment (Abacn) [inline sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaWriteSchemaTransformProvider.java:150]: what is this commented out code?

Comment (Abacn) [reply sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaWriteSchemaTransformProvider.java:150]: and the spotless check:

```
18:48:35 1: Task failed with an exception.
18:48:35 -----------
18:48:35 * What went wrong:
18:48:35 Execution failed for task ':sdks:java:io:kafka:spotlessJavaCheck'.
18:48:35 > The following files had format violations:
18:48:35       sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaWriteSchemaTransformProviderTest.java
18:48:35           @@ -20,9 +20,7 @@
18:48:35            import·java.util.Arrays;
18:48:35            import·java.util.List;
18:48:35            import·org.apache.beam.sdk.io.kafka.KafkaWriteSchemaTransformProvider.KafkaWriteSchemaTransform.ErrorCounterFn;
18:48:35           -import·org.apache.beam.sdk.io.kafka.KafkaWriteSchemaTransformProvider.KafkaWriteSchemaTransformConfiguration;
18:48:35            import·org.apache.beam.sdk.schemas.Schema;
18:48:35           -import·org.apache.beam.sdk.schemas.transforms.SchemaTransform;
18:48:35            import·org.apache.beam.sdk.schemas.utils.JsonUtils;
18:48:35            import·org.apache.beam.sdk.testing.PAssert;
18:48:35            import·org.apache.beam.sdk.testing.TestPipeline;
18:48:35   Run './gradlew :sdks:java:io:kafka:spotlessApply' to fix these violations.

```

Comment (nickuncaged1201) [reply sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaWriteSchemaTransformProvider.java]: fixed. The commented out part is the previous code before my change

**Pull request merged**